packages/grafana-sql/src/components/configuration/ConnectionLimits.tsx

```diff
@@ -115,15 +106,12 @@ export const ConnectionLimits = <T extends SQLConnectionLimits>(props: Props<T>)
           </Label>
         }
       >
-        <Input
-          type="number"
+        <NumberInput
+          value={jsonData.maxOpenConns}
+          defaultValue={0}
```

Comment:
1. i recommend changing `0` to `config.sqlConnectionLimits.maxOpenConns` .. that is the "default-value" for this field.
and then do the same for the other 2 fields too.
2. i recommend removing the placeholder-field (i know it was in the original too)... because the default-value is not always `unlimited` (actually it is usually not-never)